src/shogun/mathematics/linalg/BaseVector.h

```diff
@@ -54,7 +54,7 @@ struct BaseVector
 	}
 
 	/** Data Storage */
-	virtual bool onGPU() = 0 ;
+	virtual bool onGPU() const = 0 ;
```

Comment:
@OXPHOS what was the issue again with putting both these thing inside a single class? I think it is doable.